src/test/java/de/tum/cit/aet/artemis/shared/base/AbstractArtemisBuildAgentTest.java

```diff
@@ -0,0 +1,194 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.bean.override.mockito.MockitoSpyBean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.StartContainerCmd;
+import com.hazelcast.core.HazelcastInstance;
+
+import de.tum.cit.aet.artemis.buildagent.BuildAgentConfiguration;
+import de.tum.cit.aet.artemis.buildagent.dto.BuildConfig;
+import de.tum.cit.aet.artemis.buildagent.dto.BuildJobQueueItem;
+import de.tum.cit.aet.artemis.buildagent.dto.DockerRunConfig;
+import de.tum.cit.aet.artemis.buildagent.dto.JobTimingInfo;
+import de.tum.cit.aet.artemis.buildagent.dto.RepositoryInfo;
+import de.tum.cit.aet.artemis.buildagent.service.BuildJobGitService;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingLanguage;
+import de.tum.cit.aet.artemis.programming.domain.ProjectType;
+import de.tum.cit.aet.artemis.programming.domain.Repository;
+import de.tum.cit.aet.artemis.programming.domain.RepositoryType;
+import de.tum.cit.aet.artemis.programming.icl.DockerClientTestService;
+
+@SpringBootTest
+@ExtendWith(SpringExtension.class)
+@Execution(ExecutionMode.CONCURRENT)
+@ResourceLock("AbstractArtemisBuildAgentTest")
+// NOTE: we use a common set of active profiles to reduce the number of application launches during testing. This significantly saves time and memory!
+@ActiveProfiles({ PROFILE_TEST_BUILDAGENT, PROFILE_BUILDAGENT })
+
+// Note: the server.port property must correspond to the port used in the artemis.version-control.url property.
+@TestPropertySource(properties = { "server.port=49152", "artemis.version-control.url=http://localhost:49152", "artemis.user-management.use-external=false",
+        "artemis.version-control.local-vcs-repo-path=${java.io.tmpdir}", "artemis.build-logs-path=${java.io.tmpdir}/build-logs",
+        "artemis.continuous-integration.specify-concurrent-builds=true", "artemis.continuous-integration.concurrent-build-size=2",
+        "artemis.continuous-integration.asynchronous=true", "artemis.continuous-integration.build.images.java.default=dummy-docker-image",
+        "artemis.continuous-integration.image-cleanup.enabled=true", "artemis.continuous-integration.image-cleanup.disk-space-threshold-mb=1000000000",
+        "spring.liquibase.enabled=false", "artemis.version-control.ssh-private-key-folder-path=${java.io.tmpdir}", "artemis.version-control.build-agent-use-ssh=false",
```

Comment:
do we need to override all of those? for instance, "spring.liquibase.enabled=false" seems to be set in `src/main/resources/config/application-buildagent.yml`.

I think it would generally be good to only override that are actually (and explicitly) different to continuously ensure that the "PROD"-configs are working fine.